### PR TITLE
libtheora: fix build on macos

### PR DESCRIPTION
**Pull request by aumuell**
Fix build on macos. Applying the upstream patches with changes to `Makefile.am` require rerunning of `automake`/`autoconf`/.... The previous patch designed to prevent `configure` from being run by `autogen.sh` bailed out too early, before `automake` was invoked. This patch was dropped in favor of passing `-V` to `autogen.sh`: this option gets forwarded to `configure`, which then just prints its version and exits.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the two branches exist.

**Comment by tldahlgren** [comment]
@michaelkuhn @quellyn @johnwparent As relatively recent contributors, do any of you want to review the changes in the PR?

File: var/spack/repos/builtin/packages/libtheora/package.py [hunk 1]
```diff
@@ -43,12 +46,23 @@ class Libtheora(AutotoolsPackage, MSBuildPackage):
         "msbuild", "autotools", default="autotools" if sys.platform != "win32" else "msbuild"
     )
 
-    patch("exit-prior-to-running-configure.patch", when="@1.1.1")
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/libtheora/package.py:46]: If we don't use the patch anymore, we should also remove it from the repository.

**Review by michaelkuhn — CHANGES_REQUESTED** [review]
LGTM (except for one comment about the now-unused patch). I couldn't test the macOS-specific part, but verified that the configure change works as intended and libtheora still builds on Linux.

File: var/spack/repos/builtin/packages/libtheora/package.py (hunk 1 above)
Comment (aumuell) [reply var/spack/repos/builtin/packages/libtheora/package.py:46]: Thank you, should have thought about that.

**Review by michaelkuhn — APPROVED** [review]

**Pull request merged**